modules/dynamic/src/smithy4s/dynamic/internals/DynamicModelCompiler.scala

```diff
@@ -389,7 +484,7 @@ private[dynamic] object Compiler {
                     )
                   }
                 val memberHints = allHints(mShape.traits)
-                lField.map(_.addHints(memberHints: _*))
+                lField.map(_.addHints(memberHints.all.toSeq: _*))
```

Comment:
this is one problematic case - I wanted to add an `addHints` overload to `SchemaFieldOps`, but that breaks binary compatibility on 2.12 (the existing extension is renamed due to [weird 2.12 compiler behavior](https://github.com/scala/scala/pull/7896)). We can consider it for 0.16.0, in the meantime here's the workaround.